cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -585,6 +593,9 @@ class SimpleArray
         }
     }
 
+    template <typename I>
+    SimpleArray take_along_axis(SimpleArray<I> const & indices);
```

Comment:
It should not be too difficult to infer that `take_along_axis()` should go to `SimpleArrayMixinCalculators` with `sort()` and `argsort()` because `take_along_axis()` takes the results produced by `argsort()`.